File: web/src/services/authService.ts
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/auth';

export interface RegisterRequest {
  username: string;
  email: string;
  password: string;
}

export interface LoginRequest {
  email: string;
  password: string;
}

export interface LoginResponse {
  accessToken: string;
  tokenType: string;
  role: string;
  email: string;
  username: string;
}

export const register = async (
  data: RegisterRequest
) => {
  const response = await axios.post(
    `${API_URL}/register`, 
    data
  );
  return response.data;
};

export const login = async (
  data: LoginRequest
): Promise<LoginResponse> => {
  const response = await axios.post(
    `${API_URL}/login`, 
    data
  );
  return response.data;
};

File: backend/src/main/java/edu/cit/bien/labalink/service/AuthService.java
package edu.cit.bien.labalink.service;

import edu.cit.bien.labalink.dto.LoginRequest;
import edu.cit.bien.labalink.dto.LoginResponse;
import edu.cit.bien.labalink.dto.RegisterRequest;
import edu.cit.bien.labalink.model.User;
import edu.cit.bien.labalink.repository.UserRepository;
import edu.cit.bien.labalink.security.JwtProvider;
import lombok.RequiredArgsConstructor;
import org.springframework.security.crypto.password
    .PasswordEncoder;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class AuthService {

    private final UserRepository userRepository;
    private final PasswordEncoder passwordEncoder;
    private final JwtProvider jwtProvider;

    public void register(RegisterRequest request) {
        if (userRepository
                .existsByEmail(request.getEmail())) {
            throw new RuntimeException(
                "Email already exists");
        }

        User user = new User();
        user.setUsername(request.getUsername());
        user.setEmail(request.getEmail());
        user.setPasswordHash(
            passwordEncoder.encode(
                request.getPassword()));
        user.setRole(User.Role.CUSTOMER);

        userRepository.save(user);
    }

    public LoginResponse login(LoginRequest request) {
        User user = userRepository
                .findByEmail(request.getEmail())
                .orElseThrow(() -> 
                    new RuntimeException(
                        "Invalid credentials"));

        if (!passwordEncoder.matches(
                request.getPassword(), 
                user.getPasswordHash())) {
            throw new RuntimeException(
                "Invalid credentials");
        }

        String token = jwtProvider
            .generateToken(user.getEmail());

        return new LoginResponse(
                token,
                "Bearer",
                user.getRole().name(),
                user.getEmail(),
                user.getUsername()
        );
    }
}